src/plugins/discourse/mirrorRepository.js

```diff
@@ -45,6 +52,13 @@ export interface ReadRepository {
    * Gets all of the like actions in the history.
    */
   likes(): $ReadOnlyArray<LikeAction>;
+
+  /**
+   * Finds the TopicIds of topics that have one of the categoryIds as it's category.
+   */
+  topicsInCategories(
```

Comment:
Why do we have it takes a list of categoryIds rather than a single one? Intuitively, I would expect it to take a single categoryId and then have callers `map` over it. although having it take an array is fine too (and if we mostly want to select for groups of categories, it's a lot more convenient)

Reply:
The initial use-case is to support the mirror option
`recheckTopicsInCategories: $ReadOnlyArray<CategoryId>` which would save SQL round trips if there were multiple IDs configured. I've now made it part of ReadRepository to use for the initiatives plugin. But that is guaranteed to be just one category ID.
Neither are a hot loop though, so a singular approach would be fine too.

Reply:
I'm inclined to leave this, as this results in the least amount of SQL round trips in all situations and the overhead of using it for 1 category is to add that category to an array.